prov/usnic/src/usdf_fabric.c

```diff
@@ -101,6 +101,13 @@ usdf_validate_hints(struct fi_info *hints, struct usd_device_attrs *dap)
 		return -FI_ENODATA;
 	}
 
+	/* it doesn't make sense to request both prefix and prefix2 */
+	if ((hints->mode & (FI_MSG_PREFIX | FI_MSG_PREFIX2)) ==
+			(FI_MSG_PREFIX | FI_MSG_PREFIX2)) {
+		USDF_INFO("Simultaneous use of FI_MSG_PREFIX and FI_MSG_PREFIX2 is invalid.\n");
+		return -EINVAL;
+	}
```

Comment:
It actually makes sense for an app to request that it can support both options.  What doesn't make sense is the provider reporting that it needs both.